src/main/java/com/uber/cadence/internal/compatibility/proto/serviceclient/GrpcServiceStubs.java

```diff
@@ -80,12 +81,17 @@ final class GrpcServiceStubs implements IGrpcServiceStubs {
   private final MetaAPIGrpc.MetaAPIFutureStub metaFutureStub;
 
   GrpcServiceStubs(ClientOptions options) {
-    this.channel =
-        ManagedChannelBuilder.forAddress(options.getHost(), options.getPort())
-            .defaultLoadBalancingPolicy("round_robin")
-            .usePlaintext()
-            .build();
-
+    if (options.getGRPCChannel() != null) {
+      this.channel = options.getGRPCChannel();
+      shutdownChannel = false;
+    } else {
+      this.channel =
+          ManagedChannelBuilder.forAddress(options.getHost(), options.getPort())
+              .defaultLoadBalancingPolicy("round_robin")
```

Comment:
is `round_robbin` a constant somewhere?  Should it be?

Reply:
good point, no strong opinion here but prefer go with inline mostly